pkg/cluster/deploystorage_resources.go

```diff
@@ -74,19 +74,82 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
-func (m *manager) storageAccount(name, region string) *arm.Resource {
-	return &arm.Resource{
-		Resource: &mgmtstorage.Account{
-			Sku: &mgmtstorage.Sku{
-				Name: "Standard_LRS",
-			},
-			AccountProperties: &mgmtstorage.AccountProperties{
-				MinimumTLSVersion: mgmtstorage.TLS12,
+// storageAccount will return storage account resource.
+// Old accounts are not encrypted and can't be retrofitted.
+// flag is to controll this behaviour in update/create
+func (m *manager) storageAccount(name, region string, encrypted bool) *arm.Resource {
+	sa := &mgmtstorage.Account{
+		Kind: mgmtstorage.StorageV2,
+		Sku: &mgmtstorage.Sku{
+			Name: "Standard_LRS",
+		},
+		AccountProperties: &mgmtstorage.AccountProperties{
+			AllowBlobPublicAccess:  to.BoolPtr(false),
+			EnableHTTPSTrafficOnly: to.BoolPtr(true),
+			MinimumTLSVersion:      mgmtstorage.TLS12,
+			NetworkRuleSet: &mgmtstorage.NetworkRuleSet{
+				Bypass: mgmtstorage.AzureServices,
+				VirtualNetworkRules: &[]mgmtstorage.VirtualNetworkRule{
+					{
+						VirtualNetworkResourceID: to.StringPtr(m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID),
+						Action:                   mgmtstorage.Allow,
+					},
+					{
+						VirtualNetworkResourceID: to.StringPtr(m.doc.OpenShiftCluster.Properties.WorkerProfiles[0].SubnetID),
+						Action:                   mgmtstorage.Allow,
+					},
+					{
+						VirtualNetworkResourceID: to.StringPtr("/subscriptions/" + m.env.SubscriptionID() + "/resourceGroups/" + m.env.ResourceGroup() + "/providers/Microsoft.Network/virtualNetworks/rp-pe-vnet-001/subnets/rp-pe-subnet"),
+						Action:                   mgmtstorage.Allow,
+					},
+					{
+						VirtualNetworkResourceID: to.StringPtr("/subscriptions/" + m.env.SubscriptionID() + "/resourceGroups/" + m.env.ResourceGroup() + "/providers/Microsoft.Network/virtualNetworks/rp-vnet/subnets/rp-subnet"),
+						Action:                   mgmtstorage.Allow,
+					},
+					{
+						VirtualNetworkResourceID: to.StringPtr("/subscriptions/" + m.env.SubscriptionID() + "/resourceGroups/" + m.env.GatewayResourceGroup() + "/providers/Microsoft.Network/virtualNetworks/gateway-vnet/subnets/gateway-subnet"),
```

Comment:
Decision was to gate this gateway rule on the `env.IsLocalDevelopmentMode()`

If we're in `LocalDevelopmentMode` (CI e2e, or localRP), then don't add this rule as it will fail.